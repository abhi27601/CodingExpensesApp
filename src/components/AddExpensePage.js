import React from 'react';

const AddExpensePage = (props) => {console.log(props) 
    return(
    <div>
        This is from AddExpense component.
    </div>
)
    }

export default AddExpensePage;
