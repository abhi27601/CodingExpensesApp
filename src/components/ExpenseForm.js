import React from "react";
import "react-dates/initialize";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      note: props.expense ? props.expense.note : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarfocused: false,
      error: "",
    };
  }
  /*state={
        description:'',
        amount:'',
        note:'',
        createdAt:moment(),
        calendarfocused:false,
        error:''
    }*/
  onDescriptionchange = (value) => {
    this.setState((prevState) => ({
      description: value,
    }));
  };

  onAmountchange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState((prevState) => ({
        amount,
      }));
    }
  };
  // e.persist() if u use e.target.value in callback u ll get error for persist() //105
  onNotechange = (e) => {
    const note = e.target.value;
    this.setState((prevState) => ({ note }));
  };

  onDateChange = (createdAt) => {
    if (createdAt) this.setState({ createdAt });
  };

  onFocusChange = ({ focused }) => this.setState({ calendarfocused: focused }); // PropTypes.func.isRequired

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      const error = "Please enter proper value";
      this.setState((prevState) => ({
        error,
      }));
    } else {
      this.setState((prevState) => ({ error: "" }));
      console.log("!!Submitted !!");
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount) * 100,
        note: this.state.note,
        createdAt: this.state.createdAt.valueOf(),
      });
    }
  };

  render() {
    return (  
        <form className='form' onSubmit={this.onSubmit}>
        {this.state.error && <p className = 'form__error'>{this.state.error}</p>}
          <input
            type="text"
            placeholder="Description"
            className = 'text-input'
            autoFocus
            value={this.state.description}
            onChange={(e) => {
              this.onDescriptionchange(e.target.value);
            }}
          />

          <input
            type="text"
            placeholder="amount"
            className = 'text-input'
            value={this.state.amount}
            onChange={this.onAmountchange}
          />

          <SingleDatePicker
            date={this.state.createdAt} // momentPropTypes.momentObj or null
            onDateChange={this.onDateChange} // PropTypes.func.isRequired
            focused={this.state.calendarfocused} // PropTypes.bool
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={(day) => false}
            id="singledatepicker" // PropTypes.string.isRequired,
          />

          <textarea
            placeholder="Note for Expense (optional)"
            className = 'textarea'
            value={this.state.note}
            onChange={this.onNotechange}
          />
          
          <div>
          <button className = 'button'>Add Expense</button>
          </div>
          
        </form>
    );
  }
}
