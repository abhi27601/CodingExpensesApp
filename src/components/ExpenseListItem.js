import React from "react";
import { NavLink } from "react-router-dom";
import numeral from "numeral";
import { connect } from "react-redux";
import { removeExpense } from "../actions/expenses";
import moment from "moment";

const ExpenseListItem = (props) => (
  <NavLink className='list-item' to={`/edit/${props.id}`} activeClassName="is-active">
    <div>
      <h3 className = 'list-item__title'>{props.description}</h3>
      <span className = 'list-item__sub-title'>{moment(props.createdAt).format("MMM Do, YYYY")}</span>
    </div>
    <h3 className = 'list-item__data'> {numeral(props.amount / 100).format("$0,0.00")}</h3>
   
  </NavLink>
);
export default ExpenseListItem;
//export default connect()(ExpenseListItem)
