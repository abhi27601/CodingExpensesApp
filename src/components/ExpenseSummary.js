import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import { NavLink } from "react-router-dom";
import getTotalSelector from "../selectors/getTotalSelector";
import getVisibleExpenses from "../selectors/getVisibleExpenses";

export const ExpenseSummary = (props) => {
  const expenseWord = props.expenseCount > 1 ? "expenses" : "expense";
  const formatTotalexpense = numeral(props.Totalexpense / 100).format(
    "$0,0.00"
  );

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          {" "}
          Viewing <span> {props.expenseCount} </span> {expenseWord} and sum is{" "}
          <span>{formatTotalexpense} </span>{" "}
        </h1>
        <div className="page-header__actions">
          <NavLink to="/create" className="button">
            Add Expense
          </NavLink>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  const visibleexpenses = getVisibleExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleexpenses.length,
    Totalexpense: getTotalSelector(visibleexpenses),
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
