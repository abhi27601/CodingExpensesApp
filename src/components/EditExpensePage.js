import React from "react";
import ExpenseForm from "./ExpenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";
import { connect } from "react-redux";

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    console.log("Edit Expense Form");
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };
  onRemoveExpense = (e) => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  };

  render() {
    return (
      <div className = 'page-header'>
        <div className = 'content-container'>
          <h1 className = 'page-header__title'>Edit Expense</h1>
        {/*Eding the expense with id of {this.props.match.params.id}*/}
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button className='button button--secondary' onClick={this.onRemoveExpense}>Remove</button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  console.log(state);
  console.log(ownProps);
  return {
    expense: state.expenses.find((expense) => {
      return expense.id == ownProps.match.params.id;
    }),
  };
};
const mapDispatchToProps = (dispatch, props) => {
  return {
    startEditExpense: (id, updates) => dispatch(startEditExpense(id, updates)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
