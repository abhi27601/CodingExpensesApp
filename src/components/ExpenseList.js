import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import getVisibleExpenses from "../selectors/getVisibleExpenses";
import getTotalSelector from "../selectors/getTotalSelector";
import numeral from "numeral";
export const ExpenseList = (props) => (
  <div className = 'content-container'>
    <div className = 'list-header'>
      <div className = 'show-for-mobile' >Expenses</div>
      <div className = 'show-for-desktop' >Expense</div>
      <div className = 'show-for-desktop' >Amount</div>
    </div>
    <div className = 'list-body'>
    {props.expenses.length === 0 ? (
      
      <div className = 'list-item list-item--message'>
      <span>No expenses</span>
      </div>
    ) : (
      props.expenses.map((expense, index) => {
        return (
          <ExpenseListItem key={expense.id} {...expense}></ExpenseListItem>
        );
      })
    )}
    </div>
  </div>
);
const mapStateToProps = (state) => {
  const expensestotal = getTotalSelector(state.expenses);
  console.log(numeral(expensestotal / 100).format("$0,0.00"));
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
    //filters:state.filters not needed for this component
  };
};

export default connect(mapStateToProps)(ExpenseList);
/*
const ConnectedExpenseList = connect((state) => {
    return {
        expenses:state.expenses
    }

})(ExpenseList);*/

//export default ConnectedExpenseList;
//export default ExpenseList;
