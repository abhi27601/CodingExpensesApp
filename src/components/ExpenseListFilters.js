import React from 'react';
import ExpenseListItem from './ExpenseListItem';
import {connect  } from "react-redux";
import {setTextfilter, sortByAmount, sortByDate} from "../actions/filters";
const ExpenseListFilters = (props) => (

    <div>
        <input type='text' value={props.filters.text} onChange ={ (e) => { 
            props.dispatch(setTextfilter(e.target.value))
            console.log(e.target.value);
        }} />

        <select value={props.filters.sortBy} onChange = { (e) => {
            
            if(e.target.value == 'date'){
                props.dispatch(sortByDate());
            
            } 
            else {
                props.dispatch(sortByAmount());
            }
        }}>
            <option value='date'>Date</option>
            <option value='amount'>Amount</option>
        </select>
        
{/*
        <DateRangePicker
            startDate={this.state.startDate} // momentPropTypes.momentObj or null,
            startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
            endDate={this.state.endDate} // momentPropTypes.momentObj or null,
            endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
            onDatesChange={({ startDate, endDate }) => this.setState({ startDate, endDate })} // PropTypes.func.isRequired,
            focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
            onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
        /> */}

    </div>
)
 
const mapStateToProps = (state) => {
    return {
        filters:state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListFilters)