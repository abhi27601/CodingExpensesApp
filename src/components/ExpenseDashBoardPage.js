import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpenseSummary from "./ExpenseSummary";
const ExpenseDashBoardPage = () => (
  <div>
    <ExpenseSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>


);
console.log("Gauri");

export default ExpenseDashBoardPage;
