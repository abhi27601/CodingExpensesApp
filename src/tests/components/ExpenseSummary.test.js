import React from 'react';
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";
 
test('should correctly render ExpenseSummary',() => {
    const wrapper = shallow(<ExpenseSummary expenseCount= {1}  Totalexpense ={200} />)
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount= {2}  Totalexpense ={2000} />)
    expect(wrapper).toMatchSnapshot();
})