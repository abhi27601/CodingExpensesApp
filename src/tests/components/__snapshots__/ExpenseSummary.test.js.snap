// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       
      Viewing 
      <span>
         
        1
         
      </span>
       
      expense
       and sum is
       
      <span>
        $2.00
         
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="button"
        to="/create"
      >
        Add Expense
      </NavLink>
    </div>
  </div>
</div>
`;

exports[`should correctly render with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       
      Viewing 
      <span>
         
        2
         
      </span>
       
      expenses
       and sum is
       
      <span>
        $20.00
         
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="button"
        to="/create"
      >
        Add Expense
      </NavLink>
    </div>
  </div>
</div>
`;
